src/middleware/auth/logto.ts

```diff
@@ -239,60 +228,52 @@ export class LogToHelper {
                 Authorization: token,
             },
         });
-        const data = await response.json();
-        if (response.ok) {
-            this.m2mToken = data.access_token
+        if (!response.ok) {
+            return this.returnError(response.status, await response.text())
+            
         }
+        const data = await response.json();
+        this.m2mToken = data.access_token
+        return this.returnOk({})
     } catch (err) {
-        return {
-            status: 500,
-            error: `setM2MToken: ${err}`,
-            data: {}
-        }
+        return this.returnError(500, `setM2MToken: ${err}`)
```

Comment:
Will this be thrown to an end user ever? If yes, check whether there's a risk of the error message revealing too much about the server. If it only logs within say DO, this is fine, if it gets bubbled up to the user might be a bad idea.